console/src/main/java/org/matrix/console/Matrix.java

```diff
@@ -243,44 +244,31 @@ public synchronized void addSession(MXSession session) {
 
     /**
      * Creates an MXSession from some credentials.
-     * @param credentials The credentials to create a session from.
+     * @param hsConfig The HomeserverConnectionConfig to create a session from.
      * @return The session.
      */
-    public MXSession createSession(Credentials credentials) {
-        return createSession(mAppContext, credentials, true);
+    public MXSession createSession(HomeserverConnectionConfig hsConfig) {
+        return createSession(mAppContext, hsConfig);
     }
 
     /**
      * Creates an MXSession from some credentials.
      * @param context the context.
-     * @param credentials The credentials to create a session from.
-     * @param useHttps True to enforce https URIs on the home server.
+     * @param hsConfig The HomeserverConnectionConfig to create a session from.
      * @return The session.
      */
-    public MXSession createSession(Context context, Credentials credentials, boolean useHttps) {
-        if (!credentials.homeServer.startsWith("http")) {
-            if (useHttps) {
-                credentials.homeServer = "https://" + credentials.homeServer;
-            }
-            else {
-                credentials.homeServer = "http://" + credentials.homeServer;
-            }
-        }
-
-        // remove the trailing /
-        if (credentials.homeServer.endsWith("/")) {
-            credentials.homeServer = credentials.homeServer.substring(0, credentials.homeServer.length()-1);
-        }
-
+    public MXSession createSession(Context context, HomeserverConnectionConfig hsConfig) {
         IMXStore store;
 
-        if (true ) {
-            store = new MXFileStore(credentials, context);
+        Credentials credentials = hsConfig.getCredentials();
+
+        if (true) {
+            store = new MXFileStore(hsConfig, context);
         } else {
-            store = new MXMemoryStore(credentials);
+            store = new MXMemoryStore(hsConfig.getCredentials());
```

Comment:
Foreword: I know you just added an arg here and didn't write the `if` statement.

I *assume* this is for debugging? This should be commented as such (or even better, a self-documenting `SHOUTY_CAPS_CONSTANT`).